setup_new_plugin.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/sh
+
+# This script sets up a new plugin with the input name
+#
+# Usage:
+#   setup_new_plugin.sh [plugin_name]
+#
+
+###################
+## Plugin Basics ##
+###################
+set -e # Exit with error if any command fails
+
+if [ -z "$1" ]
+then
+  read -p "Enter the plugin library name: " plugin_name
+else
+  plugin_name=$1
+fi
+
+## plugin_name.txt
+cat << EOF > plugin_name.txt
+${plugin_name}
+EOF
+
+## version.txt
+cat << EOF > version.txt
+0.0.0
+EOF
+
+## toolchain.cmake
+cat << EOF > toolchain.cmake
+set(BUILD_TESTING ON)
+set(BUILD_SHARED_LIBS ON)
+set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+set(NWX_MODULE_DIRECTORY ./install/python)
+EOF
+
+###################
+## include files ##
+###################
+mkdir -p include/${plugin_name}
+
+## include/${plugin_name}/${plugin_name}_mm.hpp
+cat << EOF > include/${plugin_name}/${plugin_name}_mm.hpp
+#pragma once
+#include <pluginplay/module_manager.hpp>
+
+namespace ${plugin_name} {
+
+/** @brief Loads the modules contained in the plugin into the provided 
+ *         ModuleManager instance.
+ */
+void load_modules(pluginplay::ModuleManager& mm);
```

Comment:
```suggestion
 */
DECLARE_PLUGIN(${plugin_name});
```

FYI this is the official API for declaring a Plugin (defined [here](https://github.com/NWChemEx/PluginPlay/blob/master/include/pluginplay/plugin/plugin.hpp)). The idea is we don't want to be tied to the `load_modules` function in case we want to switch to a class with meta data at some point.